### PR TITLE
[REM] osutil: deprecated code

### PR DESCRIPTION
**Pull request by xmo-odoo**
We deprecated all those in 14.0, so 16.0 seems like a good time to remove them.

`getppid` was not deprecated but it doesn't seem useful to keep around: windows support was added to getppid in Python 3.2.

### CODE REVIEW

File: odoo/tools/misc.py [hunk 1]
```diff
@@ -79,12 +79,6 @@ def _exec_pipe(prog, args, env=None):
     pop = subprocess.Popen(cmd, bufsize=-1, stdin=subprocess.PIPE, stdout=subprocess.PIPE, close_fds=close_fds, env=env)
     return pop.stdin, pop.stdout
 
-def exec_command_pipe(name, *args):
```
Comment (ryv-odoo) [inline odoo/tools/misc.py:82]: Doesn't seems deprecated ?

File: odoo/tools/osutil.py [hunk 2]
```diff
@@ -54,34 +54,6 @@ def clean_filename(name, replacement=''):
         return "Untitled"
     return re.sub(r'[^\w_.()\[\] -]+', replacement, name).lstrip('.-') or "Untitled"
 
-def listdir(dir, recursive=False):
```
Comment (ryv-odoo) [inline odoo/tools/osutil.py:57]: It seems still used by  `get_module_filetree` (used by `download_iot_handlers`) :/

Comment (xmo-odoo) [reply odoo/tools/osutil.py:57]: Yep yer right, I'd only deprecated the non-recursive version so I should only remove that bit.

And probably deprecate `get_module_filetree` (and recursive `listdir`), and rewrite `download_iot_handlers` to stop using that. Which means I need to understand what this mess does.

:crying_cat_face:

File: odoo/tools/misc.py (hunk 1 above)
Comment (xmo-odoo) [reply odoo/tools/misc.py:82]: Yep that's a mistake, I amended the commit with other changes.

**Comment by xmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**